modules/mantisBidAdapter.js

```diff
@@ -1,228 +1,221 @@
-
-      return !supportsTouch && (!width || width >= (window.mantis_breakpoint || 768));
-    };
-
-    if (inIframe()) {
-      try {
-        return scope(window.top);
-      } catch (ex) {
-      }
-    }
-
-    return scope(window);
+function isDesktop(ignoreTouch) {
+  var supportsTouch = !ignoreTouch && ('ontouchstart' in window || navigator.msMaxTouchPoints);
+  if (inIframe()) {
+    return !supportsTouch;
   }
+  var width = window.innerWidth || window.document.documentElement.clientWidth || window.document.body.clientWidth;
+  return !supportsTouch && (!width || width >= (window.mantis_breakpoint || 768));
+}
 
-  function isSendable(val) {
-    if (val === null || val === undefined) {
-      return false;
-    }
-
-    if (typeof val === 'string') {
-      return !(!val || /^\s*$/.test(val));
-    }
-
-    if (typeof val === 'number') {
-      return !isNaN(val);
+function storeUuid(uuid, sync) {
+  if (window.mantis_uuid) {
+    return;
+  }
+  window.mantis_uuid = uuid;
+  if (window.localStorage) {
+    try {
+      window.localStorage.setItem('mantis:uuid', uuid);
+    } catch (ex) {
     }
-
-    return true;
   }
-
-  function isObject(value) {
-    return Object.prototype.toString.call(value) === '[object Object]';
+  if (!sync) {
+    postMessage('uuid', uuid);
   }
+}
 
-  function isAmp() {
-    return typeof window.context === 'object' && (window.context.tagName === 'AMP-AD' || window.context.tagName === 'AMP-EMBED');
+function postMessage(type, data, scope, origin) {
+  var target = scope || window.top;
+  if (target.postMessage) {
+    target.postMessage({
+      mantis: true,
+      type: type,
+      data: data
+    }, origin || '*');
   }
+}
 
-  function isSecure() {
-    return document.location.protocol === 'https:';
+function isSendable(val) {
+  if (val === null || val === undefined) {
+    return false;
   }
-
-  function isArray(value) {
-    return Object.prototype.toString.call(value) === '[object Array]';
+  if (typeof val === 'string') {
+    return !(!val || /^\s*$/.test(val));
   }
-
-  function jsonp(callback) {
-    if (!window.mantis_jsonp) {
-      window.mantis_jsonp = [];
-    }
-
-    window.mantis_jsonp.push(callback);
-
-    return 'mantis_jsonp[' + (window.mantis_jsonp.length - 1) + ']';
+  if (typeof val === 'number') {
+    return !isNaN(val);
   }
+  return true;
+}
 
-  function jsonToQuery(data, chain, form) {
-    if (!data) {
-      return null;
-    }
-
-    var parts = form || [];
-
-    for (var key in data) {
-      var queryKey = key;
+function isObject(value) {
+  return Object.prototype.toString.call(value) === '[object Object]';
+}
 
-      if (chain) {
-        queryKey = chain + '[' + key + ']';
-      }
+function isAmp() {
+  return typeof window.context === 'object' && (window.context.tagName === 'AMP-AD' || window.context.tagName === 'AMP-EMBED');
+}
 
-      var val = data[key];
+function isSecure() {
+  return document.location.protocol === 'https:';
+}
 
```

Comment:
Note: There is a function for testing types included in [utils.js](https://github.com/prebid/Prebid.js/blob/master/src/utils.js) beginning on line #332. Line #334 includes an isArray method where you can see how the "isA" method could be used for testing types not included such as "[Object]".